src/math/p5.Vector.js

```diff
@@ -584,22 +602,31 @@ p5.Vector.prototype.sub = function sub(x, y, z) {
  * </code>
  * </div>
  */
-p5.Vector.prototype.mult = function mult(n) {
-  if (!(typeof n === 'number' && isFinite(n))) {
-    console.warn(
-      'p5.Vector.prototype.mult:',
-      'n is undefined or not a finite number'
-    );
+p5.Vector.prototype.mult = function mult(x, y, z) {
+  if (x instanceof p5.Vector) {
+    this.x *= x.x || 0;
+    this.y *= x.y || 0;
+    this.z *= x.z || 0;
     return this;
   }
-  this.x *= n;
-  this.y *= n;
-  this.z *= n;
+  if (x instanceof Array) {
```

Comment:
Do we need to cover the case where array is passed in as well?

Reply:
Again, also following the pattern laid out in .add and .sub, which both also check if an array was passed in.

Reply:
@limzykenneth updated!